packages/protocol/contracts/stability/PairManager.sol

```diff
@@ -0,0 +1,290 @@
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    setBroker(_broker);
+  }
+
+  /* ==================== Modifiers ==================== */
+
+  modifier onlyBroker() {
+    require(msg.sender == broker, "Caller is not the Broker");
+    _;
+  }
+
+  /* ==================== View Functions ==================== */
+
+  function getPair(bytes32 pairId) public view returns (Pair memory pair) {
+    require(isPair[pairId], "A pair with the specified id does not exist");
+    pair = pairs[pairId];
+  }
+
+  /* ==================== Mutative Functions ==================== */
+
+  /**
+   * @notice Sets the address of the broker contract.
+   * @param _broker The new address of the broker contract.
+   */
+  function setBroker(address _broker) public onlyOwner {
+    require(_broker != address(0), "Broker address must be set");
+    broker = _broker;
+    emit BrokerUpdated(_broker);
+  }
+
+  /**
+   * @notice Creates a new pair using the given parameters.
+   * @param pair The information required to create the pair.
+   * @return pairId The id of the newly created pair.
+   */
+  function createPair(Pair calldata pair) external onlyOwner returns (bytes32 pairId) {
+    require(address(pair.mentoExchange) != address(0), "Mento exchange must be set");
+    require(address(pair.stableAsset) != address(0), "Stable asset must be set");
+    require(address(pair.collateralAsset) != address(0), "Collateral asset must be set");
+
+    Pair memory pairInfo = pair;
+
+    pairId = keccak256(
+      abi.encodePacked(
+        IERC20Metadata(pairInfo.stableAsset).symbol(),
+        IERC20Metadata(pairInfo.collateralAsset).symbol(),
+        pairInfo.mentoExchange.name()
+      )
+    );
+    require(!isPair[pairId], "A pair with the specified assets and exchange exists");
+
+    validatePairInfo(pairInfo);
+
+    // TODO: getUpdatedBuckets Function sig should change. May not need amounts
+    // but instead pass minSupplyForStableBucketCap & stableBucketMaxFraction
+    (uint256 tokenInBucket, uint256 tokenOutBucket) = pairInfo.mentoExchange.getUpdatedBuckets(
+      pairInfo.stableAsset,
+      pairInfo.collateralAsset,
+      0,
+      0,
+      pairId
+    );
+
+    pairInfo.stableBucket = tokenInBucket;
+    pairInfo.collateralBucket = tokenOutBucket;
+
+    // TODO: https://github.com/mento-protocol/mento-general/issues/60
+    // Update allocated collateral fraction
+    // allocatedCollateralFractions[pairInfo.collateralAsset] = allocatedCollateralFractions[
+    //   pairInfo.collateralAsset
+    // ].add(pairInfo.collateralBucketFraction);
+
+    // // Update allocated stable bucket fraction
+    // allocatedStableFractions[pairInfo.stableAsset] = allocatedStableFractions[pairInfo.stableAsset]
+    //   .add(pairInfo.stableBucketMaxFraction);
+
+    pairs[pairId] = pairInfo;
+    isPair[pairId] = true;
+
+    emit PairCreated(
+      pairInfo.stableAsset,
+      pairInfo.collateralAsset,
+      address(pairInfo.mentoExchange)
```

Comment:
We should add the `pairId` in the event as well, it will be useful with The Graph which I'd love to use.